core/src/main/java/org/infinispan/stream/impl/DistributedCacheStream.java

```diff
@@ -608,9 +609,9 @@ private void performLocalRehashAwareOperation(KeyTrackingConsumer<Object, R> res
          results.onCompletion(null, segments, localValue);
       } else {
          Set<Integer> ourSegments = ownedSegmentsSupplier.get();
-         ourSegments.retainAll(segmentsToProcess);
-         log.tracef("CH changed - making %s segments suspect for identifier %s", ourSegments, id);
-         results.onSegmentsLost(ourSegments);
+         Set<Integer> lostSegments = ourSegments.stream().filter(segmentsToProcess::contains).collect(Collectors.toSet());
```

Comment:
This class has no trace variable yet.  I am thinking it would be best to handle that in another JIRA since I don't want to change the entire file.